### PR TITLE
Feature filter

### PR DESCRIPTION
**Pull request by Meowmir**
## Describe your changes
Created a function to filter through given collection and return elements that passes the given requirement.

## Code Usage examples
``` js
const array = [1, 2, 3, 4, 'A']

filter(array, isEven)
// console.log([2, 4])

filter(array, isNaN)
// console.log(['A'])

filter(array, (i) => i > 3)
// console.log([4])
```

## Checklist before requesting a review
- [x] I have added JsDoc comments to all public methods created
- [x] EsLint is passing for this PR
- [x] Tests are passing for this PR
- [x] CI build is passing for this PR
- [x] I have performed a self-review of my code
- [x] I have had another developer review my code

### CODE REVIEW

File: filter.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+/**
+ *
+ * @param {array|object} collection - The collection to filter.
+ * @param {function} requirement - The requirement to filter with.
+ * @returns {array} - Returns new array with elements passing the requirement provided for each iteration.
+ */
+
+export default function filter(collection, requirement) {
+  return collection.filter((n) => requirement(n))
```
Comment (Mr-Squirrel) [inline filter.js]: Would be nice to add a guard for isArray

**Review by Mr-Squirrel — DISMISSED** [review]
Minor comment, LGTM

File: filter.js (hunk 1 above)
Comment (Meowmir) [reply filter.js]: indeed. look at new commit

File: filter.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+ * @param {function} requirement - The requirement to filter with.
+ * @returns {array} - Returns new array with elements passing the requirement provided for each iteration.
+ */
+
+export default function filter(collection, requirement) {
+  if (Array.isArray(collection)) {
+    return collection.filter((n) => requirement(n))
+  }
+  const objToArray = Object.values(collection)
+  return objToArray.filter((n) => requirement(n))
```
Comment (Mr-Squirrel) [inline filter.js:13]: Whilst I really like seeing you become more comfortable with iterables this doesn't make sense to me

`filter` is intended for arrays, not objects. I would check if not array and return empty array.

What you are doing with `Object.values` is more akin to [_.omitBy](https://lodash.com/docs/4.17.15#omitBy)

File: test/filter.test.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    assert.deepStrictEqual(filter(array, (i) => i > 3), [4])
+    assert.deepStrictEqual(filter(array, isNaN), ['A'])
+  })
+})
+
+describe('filter', () => {
+  const obj = { 1 : 'A', 2 : 'B', 3 : 'C', 4 : 5 }
+
+  it('should work with an object', () => {
+    assert.deepStrictEqual(filter(obj, (s) => typeof s === typeof 'string'), ['A', 'B', 'C'])
```
Comment (Mr-Squirrel) [inline test/filter.test.js:19]: the semantics here get weird. why on the values, why not on keys, who not on both? See my other comment

**Review by Mr-Squirrel — CHANGES_REQUESTED** [review]

File: filter.js (hunk 2 above)
Comment (Meowmir) [reply filter.js:13]: Lodash filter is for arrays and objects:
https://lodash.com/docs/4.17.15#filter

File: test/filter.test.js (hunk 3 above)
Comment (Meowmir) [reply test/filter.test.js:19]: I agree, but I didn't write this test.

**Review by Mr-Squirrel — APPROVED** [review]

**Pull request merged**